lambda.tf

```diff
@@ -0,0 +1,63 @@
+locals {
+  code_version  = var.autoscaler_version
+  function_name = "ec2-autoscaler-${var.worker_pool_id}"
+}
+
+resource "null_resource" "download" {
+  provisioner "local-exec" {
+    command = "${path.module}/download.sh ${local.code_version}"
+  }
+}
+
+data "archive_file" "binary" {
+  type        = "zip"
+  source_file = "${var.local_path}ec2-workerpool-autoscaler_v${var.autoscaler_version}"
+  output_path = "ec2-workerpool-autoscaler_v${var.autoscaler_version}.zip"
+  depends_on  = [ null_resource.download ]
+}
+
+resource "aws_lambda_function" "autoscaler" {
+  filename         = data.archive_file.binary.output_path
+  source_code_hash = data.archive_file.binary.output_base64sha256
+  function_name    = local.function_name
+  role             = aws_iam_role.lambda.arn
+  handler          = "ec2-workerpool-autoscaler_v${var.autoscaler_version}"
+
+  runtime                        = "go1.x"
+
+  environment {
+    variables = {
+      AUTOSCALING_GROUP_ARN         = module.asg.autoscaling_group_arn
+      AUTOSCALING_REGION            = data.aws_region.this.name
+      SPACELIFT_API_KEY_ID          = var.spacelift_api_key_id
+      SPACELIFT_API_KEY_SECRET_NAME = aws_ssm_parameter.spacelift_api_key_secret.name
+      SPACELIFT_API_KEY_ENDPOINT    = var.spacelift_url
+      SPACELIFT_WORKER_POOL_ID      = var.worker_pool_id
+    }
+  }
+
+  tracing_config {
+    mode = "Active"
+  }
+
+  depends_on = [ module.asg, null_resource.download ]
+}
+
+resource "aws_cloudwatch_event_rule" "scheduling" {
+  name                = "spacelift-${var.worker_pool_id}-scheduling"
+  description         = "Spacelift autoscaler scheduling for worker pool ${var.worker_pool_id}"
+  schedule_expression = "rate(1 minute)"
```

Comment:
That should probably be configurable.

Reply:
this event rule doesn't work with variable parameters unfortunately

Reply:
Are you saying that the following would not work?

```
schedule_expression = var.schedule_expression
```

The `schedule_expression` variable would default to `rate(1 minute)` but that would also allow people to tweak that.